igvm_c/README.md

```diff
@@ -0,0 +1,59 @@
+# IGVM C API
+
+This folder contains a Makefile project that is used to build a C compatible API
+from the igvm crate. The C API provides functions that can be used to parse and
+validate a binary IGVM file and provides access to the fixed header and all of
+the variable headers and associated file data.
+
+The C API is generated directly from the rust source files. This includes the
+definitions of the structures and enums in igvm_defs. This ensures that the C
+API does not need to be manually updated inline with any changes to the rust
+definitions.
+
+## Dependencies
+The C API header files are generated using the `cbindgen` tool. This tool needs
+to be installed before the API can be built. This can be achieved using:
+
+```bash
+cargo install --force cbindgen
+```
+
+In addition, `sample/dump_igvm` requires `gcc` to be installed.
```

Comment:
Could we use CC instead of requiring GCC here so you could use clang/etc? Or whatever the Makefile equivalent is, not an expert here

Reply:
Done